File: src/dotnet/HelloMutation.Domain/Entities/Review.cs
using System;

namespace HelloMutation.Domain.Entities
{
    public record Review
    {
        public Review(ushort rating, string note) => (Rating, Note) = (rating, note);

        private ushort _rating;
        public ushort Rating
        {
            get => _rating;
            init => _rating = value <= 5 ? value : throw new ArgumentOutOfRangeException(nameof(Rating));
        }
        public string Note { get; }
    }
}

File: src/javascript/src/domain/entities/review.ts
export class Review {
    private _rating!: number
    public get rating(): number {return this._rating}

    private _note!: string
    public get note(): string {return this._note}

    constructor(init: Partial<Review>) {
        this._note = init.note!
        this._rating = init.rating!
    }
}